assets/js/base/context/cart-checkout/payment-methods/constants.js

```diff
@@ -17,14 +17,18 @@ export const ACTION_TYPES = {
 	SET_REGISTERED_PAYMENT_METHODS: 'set_registered_payment_methods',
 	SET_REGISTERED_EXPRESS_PAYMENT_METHODS:
 		'set_registered_express_payment_methods',
+	SET_SHOULD_SAVE_PAYMENT_METHOD: 'set_should_save_payment_method',
```

Comment:
I managed to get a long way without defining my new action constant here – I think I was switching on `undefined` in the reducer (🙀). Any ideas how we can make this more bulletproof?

Maybe in other files we import `ACTION_TYPES` and then use full `ACTION_TYPES.bla`, or perhaps we have a module just for action types, with each exported individually.

Reply:
> Maybe in other files we import ACTION_TYPES and then use full ACTION_TYPES.bla, or perhaps we have a module just for action types, with each exported individually.

The reducer _does_ import `ACTION_TYPES` and then destructure to use. I'm not sure how what you are describing would solve the problem you experienced? One reason for setting things up the way they are now is to clarify the dependencies via imports. I'm open to ideas but I'm not sure how you can make that more "bulletproof" against incorrect implementation?